Update release notes for 2.2.0-rc1 (#1663)
theforeman/theforeman.org: _includes/manuals/2.2/1.2_release_notes.md

* Problem when assigning quota in ovirt ([#30175](https://projects.theforeman.org/issues/30175))
* fix rails 6 deprecation warnings.  ([#29716](https://projects.theforeman.org/issues/29716))
* Creating invalid image within compute resource is possible ([#29702](https://projects.theforeman.org/issues/29702))
* Foreman lists unrelated RHV storage domains using v4 API ([#29283](https://projects.theforeman.org/issues/29283))

#### DB migrations
* make Plugin#pending_migration check plugin migrations instead of core ones ([#29498](https://projects.theforeman.org/issues/29498))

#### DHCP
* iPXE Embedded loader does not put token to the DHCP filename option ([#27879](https://projects.theforeman.org/issues/27879))

#### Development tools
* update container builds to latest base image / ruby /nodejs ([#30499](https://projects.theforeman.org/issues/30499))

#### Facts
* Extract import_facts entry point from host class and create HostFactImporter ([#30127](https://projects.theforeman.org/issues/30127))
* Use facet features in reported data facet ([#30126](https://projects.theforeman.org/issues/30126))

> #### Foreman modules

Reviewer: I've often wondered if we should prefix this categories from subprojects with the subproject. That would be `Installer - Foreman modules` instead.

Author: good idea! I'll add that.

* proxying pulp rpm/isos to pulpcore with pulpcore disabled disables pulp2 plugins ([#30211](https://projects.theforeman.org/issues/30211))
* Disable TFTP by default ([#30198](https://projects.theforeman.org/issues/30198))
* Automatically clean up candlepin event queues ([#30135](https://projects.theforeman.org/issues/30135))
* Manage configuration of default_hostgroup plugin ([#30078](https://projects.theforeman.org/issues/30078))
* support configuring number of pulp3 workers ([#30057](https://projects.theforeman.org/issues/30057))

#### Host creation
* PuppetMaster and Puppet CA Language on Host Edit screen is misleading ([#30351](https://projects.theforeman.org/issues/30351))
* provide host enrollment feature ([#26925](https://projects.theforeman.org/issues/26925))

#### Host groups
* creating host from hg doesn't resolves the user-data template ([#30526](https://projects.theforeman.org/issues/30526))

#### Installer
* Add foreman_statistics plugin ([#30553](https://projects.theforeman.org/issues/30553))
* Install selinux packages via a hook ([#30506](https://projects.theforeman.org/issues/30506))

> * Split keystore and truststore to different files ([#30346](https://projects.theforeman.org/issues/30346))

Reviewer: I've updated this to include `Candlepin` in the name.

* Move bootstrap RPM creation to puppet-foreman_proxy_content ([#30316](https://projects.theforeman.org/issues/30316))
* Drop unsupported deployments from rhsm-reconfigure script ([#30312](https://projects.theforeman.org/issues/30312))
* On EL8, don't enable Pulp 2 ([#30245](https://projects.theforeman.org/issues/30245))
* Remove --clear-puppet-environments due to Pulp 3 ([#29800](https://projects.theforeman.org/issues/29800))

> * Merge --reset-foreman-db and --reset ([#29797](https://projects.theforeman.org/issues/29797))

Reviewer: Also updated to add `into --reset-data`

* Use PageLayout component on HW models page ([#26772](https://projects.theforeman.org/issues/26772))

#### Notifications
* Only require needed parts of rss gem ([#29881](https://projects.theforeman.org/issues/29881))

#### Organizations and Locations
* location/organization name is translated when using different locale and an existing string as name ([#30382](https://projects.theforeman.org/issues/30382))

#### Packaging
* During db:migrate with Sprockets 4+, Sprockets::Railtie::ManifestNeededError: Expected to find a manifest file in \`app/assets/config/manifest.js ([#30167](https://projects.theforeman.org/issues/30167))
* Migrate from awesome_print to amazing_print ([#29870](https://projects.theforeman.org/issues/29870))
* Fix activesupport/rdoc dep issue ([#15628](https://projects.theforeman.org/issues/15628))

#### Plugin integration
* ReportScanner should get all raw report data in case those could be useful. ([#30408](https://projects.theforeman.org/issues/30408))

#### Puppet integration
* Remove fixture usage from Classification testing ([#30401](https://projects.theforeman.org/issues/30401))

> * Dropping puppetrun ([#29993](https://projects.theforeman.org/issues/29993))

Reviewer: Reworded

* Provide a way for giving rake tasks the ability to act as Dynflow clients ([#30146](https://projects.theforeman.org/issues/30146))

#### Reporting
* 'Host - compare content hosts packages' report: "Safemode doesn't allow to access 'version' on #&lt;Safemode::ScopeObject&gt;" ([#30510](https://projects.theforeman.org/issues/30510))
* load_hosts macro duplicated in template help ([#30196](https://projects.theforeman.org/issues/30196))
* hide host selector from report templates. ([#28986](https://projects.theforeman.org/issues/28986))

#### SELinux
* Build a bootdisk iso - Failed to format the ESP image via mkfs.msdos ([#30422](https://projects.theforeman.org/issues/30422))

#### Search
* os_minor &lt; 0 search shows hosts with empty minor versions ([#30375](https://projects.theforeman.org/issues/30375))
* Unable to search by value of certain Hostgroup parameter ([#29683](https://projects.theforeman.org/issues/29683))
* Web UI should not show "Full Trace" when someone provides invalid search input ([#28384](https://projects.theforeman.org/issues/28384))

#### Settings

> * Allow setting full name to be changed ([#30152](https://projects.theforeman.org/issues/30152))

Reviewer: Also slightly reworded

* Update foreman fonts to patternfly v4 ([#29331](https://projects.theforeman.org/issues/29331))
* ERF42-4995 - Invalid authenticity token (Session timed out at login screen) ([#10577](https://projects.theforeman.org/issues/10577))

#### foreman-installer script
* make it possible to brand katello-certs-check ([#30251](https://projects.theforeman.org/issues/30251))
* Remove MongoDB Storage Engine Migration Hook ([#30006](https://projects.theforeman.org/issues/30006))
* Remove the dependency on foreman-maintain ([#29963](https://projects.theforeman.org/issues/29963))
* Drop --upgrade flag from katello hooks ([#29296](https://projects.theforeman.org/issues/29296))

#### foreman-tasks
* During smart proxy sync display the filter is set to "resource_id = 3 AND resource_type = SmartProxy", the tasks are displayed twice and Foreman Task id is set to "zero" ([#30376](https://projects.theforeman.org/issues/30376))

*A full list of changes in 2.2.0 is available via [Redmine](https://projects.theforeman.org/issues?set_filter=1&sort=id%3Adesc&status_id=closed&f[]=cf_12&op[cf_12]=%3D&v[cf_12][]=1234)*

### Deprecations

### Upgrade warnings

> * The `--upgrade` installer flag has been removed. When running the installer as part of the upgrade procedure it is not required.

Reviewer: The `--upgrade` flag never existed in regular Foreman. This is just something that Katello used to do. Adding it here will only confuse Foreman users.

> * Trends and statistics are now moved to a separate plugin. See the [manual](/plugins/foreman_statistics) of the new Statistics plugin if you wish continue using them.

Reviewer: And if I don't want to continue using them? Do I need to do anything?

Author: i'll add that once we have the clean up rake task ready, it didn't make the cut for rc1

* TFTP integration is now [disabled by default](https://community.theforeman.org/t/disable-foreman-proxy-tftp-support-by-default/18892). Upgrades are unaffected. To enable TFTP integration, `--foreman-proxy-tftp true` must be passed to the installer command.

> * Puppet run button has been removed from the host detail page. To trigger one-time puppet agent runs, please use the Remote Execution plugin instead. ([More info](https://community.theforeman.org/t/dropping-run-puppet-button-with-a-replacement-using-remote-execution-plugin/12781/25))

Reviewer: Isn't it `Run Puppet`? It also used to be on the host list in the bulk actions, not just on the detail page.
```suggestion
* The Run Puppet functionality based on puppetrun has been removed. To trigger one-time puppet agent runs, please use the Remote Execution plugin instead. ([More info](https://community.theforeman.org/t/dropping-run-puppet-button-with-a-replacement-using-remote-execution-plugin/12781/25))
```

* Previously the default installation set the `destroy_vm_on_host_delete` setting to `true`, meaning associated VMs were automatically destroyed upon host deletion in Foreman. This caused a lot of pain for users who did not expect such behavior, and mistakenly destroyed production machines when attempting to remove them from Foreman. The default behavior has been changed and VMs will now remain running on the hypervisor or in the cloud. Notice that this can lead to unexpected payments in case you are paying for these VMs. If you prefer to destroy VMs automatically upon host deletion, switch this setting back to `true`.
* The template helper named `foreman_url` now requires an argument which represents which template kind to render. Previously, no argument was needed to render `provision` template kind. All default templates were updated, for custom templates please rewrite `foreman_url` to `foreman_url('provision')`.

> ### Contributors

Reviewer: Looks like you didn't run this part yet

Author: right, i need the ga tag for the contributor script to gather the names. I do that as part of the GA update.
